File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/modbus/RegisterBase.java
package com.dfi.sbc2ha.config.sbc2ha.definition.modbus;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class RegisterBase {

    int base;
    int length;
    List<Register> registers = new ArrayList<>();
    @JsonProperty("register_type")
    ModbusRegisterType modbusRegisterType = ModbusRegisterType.INPUT;

}

File: config/editor/src/app/definition/sensor/modbus/RegisterBase.ts
import {Register} from "./Register";
import {ModbusRegisterType} from "../../enums/ModbusRegisterType";

export class RegisterBase {
    base: number;
    length: number;
    registers: Register[]
    modbusRegisterType: ModbusRegisterType = ModbusRegisterType.INPUT;
}
